_posts/blog/2016-06-22-vector_drawables.md

```diff
@@ -0,0 +1,112 @@
+
+{% highlight java %}
+@Override
+protected void onDraw(final Canvas canvas) {
+    long startTime = System.currentTimeMillis();
+
+    super.onDraw(canvas);
+
+    long endTime = System.currentTimeMillis();
+    notifyDraw(startTime, endTime);
+}
+{% endhighlight %}
+
+Tests were done one a Samsung Galaxy S7 device.
+
+Test 1:
+Image size - 1440px x 1960px
+Duration:
+- vector drawable: 16 ms
+- PNG: 0ms
+
+Test 2:
+Image size - 400px x 400px
+Duration:
+- vector drawable: 2 ms
+- PNG: 0ms
+
+
+
+When vector drawables are drawn for the first time, a cached bitmap is created in order to optimize the re-drawing performance. This cache is re-used as long as the width and the height of the image that needs to be drawn is the same. If a VectorDrawable is used for multiple sizes, a new Bitmap will be created every time and drawn. Therefore, the best approach in this case is to create VectorDrawables for every size needed.
+
+In order to test this, we just allow our app to support both portrait and landscape orientation. Since the size of the view changes when changing the orientation, we can see that when rotating the device, drawing the vector drawable takes 15ms in portrait and 5ms in landscape, where the view is smaller.
+In the same time, when the image is just re-used by different views, we can see that drawing the vector drawable takes 15ms only the first time, afterwards, drawing takes 0ms.  
+
+
+### Supported tags
+
+Not all VectorDrawable XML tags are supported for Android 5.0 and higher, but just a limited set:
+
+// insert image
+
+But, the Support Library 23.2.0 and higher is offering full support for VectorDrawable XML elements.
+
+
+### Resource references in VectorDrawable XML
+
+In case you want to support things like this in the VectorDrawable XML:
+
+{% highlight XML %}
+android:fillColor="@color/colorPrimary"
+{% endhighlight %}
+
+Be aware that only Android 5.0 and higher supports dynamic attributes.
+Setting this will result in black areas instead of your desired color, on pre-API level 21 devices.
+
+<center>
+<picture class="half">
+	<img src="/images/blog/vector_drawables/dynamic_res_21.png" alt="Dynamic attributes on API level 21">
+	<figcaption>Dynamic attributes used on API level 21</figcaption>
+</picture>
+</center>
+
+<center>
+<picture>
+	<img src="/images/blog/vector_drawables/dynamic_res_19.png" alt="Dynamic attributes on API level 19">
+	<figcaption>Dynamic attributes used on API level 19</figcaption>
+</picture>
+</center>
+
+### Vector Drawable preview in Android Studio looks skewed
+
+The Vector Drawable preview either in the import preview or in the file preview might look skewed. But, don't be scared, chances are it might look good when the app is run on the device. So, before panicking or even worse, giving up check it first on the device.
```

Comment:
Maybe change "So, before panicking or even worse, giving up check" change to "So, before panicking or even worse **- giving up, check**"